java-util/src/main/java/org/apache/druid/java/util/http/client/response/HttpResponseHandler.java

```diff
@@ -38,17 +38,60 @@
  *
  * Note: if you return a finished ClientResponse object from anything other than the done() method, IntermediateType
  * must be castable to FinalType
+ *
+ * This handler can exert backpressure by returning a response with "continueReading" set to false from handleResponse()
+ * or handleChunk(). In this case, the HTTP client will stop reading soon thereafter. It may not happen immediately, so
+ * be prepared for more handleChunk() calls to happen. To resume reads, call resume() on the TrafficCop provided by
```

Comment:
is this because netty `channel.setReadable(false)` doesn't take effect immediately and netty might deliver one or few more chunks after changing channel readability?